temporal-sdk/src/main/java/io/temporal/internal/sync/SignalDispatcher.java

```diff
@@ -52,6 +48,8 @@ class SignalDispatcher {
   /** Buffers signals which don't have a registered listener. */
   private final Queue<SignalData> signalBuffer = new ArrayDeque<>();
 
+  private Map<Long, SignalHandlerInfo> runningSignalHandlers = new TreeMap<>();
```

Comment:
A linked hash map would also work here, but this is fine too. May want to make LHS be `SortedMap` for readers though, but no biggie.